src/transaction.ts

```diff
@@ -134,40 +134,80 @@ export function parseRawSimulation(
     // Gordon Ramsey in shambles
     return sim;
   }
-  return {
+
+  // shared across all responses
+  let base = {
     id: sim.id,
-    minResourceFee: sim.minResourceFee,
     latestLedger: sim.latestLedger,
-    cost: sim.cost,
-    transactionData: new SorobanDataBuilder(sim.transactionData),
-    events: (sim.events ?? []).map((event) =>
-      xdr.DiagnosticEvent.fromXDR(event, "base64")
-    ),
-    ...(sim.error !== undefined && { error: sim.error }), // only if present
-    // ^ XOR v
-    ...((sim.results ?? []).length > 0 && {
-      result: sim.results?.map((result) => {
-        return {
-          auth: (result.auth ?? []).map((entry) =>
-            xdr.SorobanAuthorizationEntry.fromXDR(entry, "base64")
-          ),
-          retval: xdr.ScVal.fromXDR(result.xdr, "base64"),
-        };
-      })[0], // only if present
-    }),
+    events: sim.events?.map(
+      evt => xdr.DiagnosticEvent.fromXDR(evt, 'base64')
+    ) ?? [],
   };
+
+  return (typeof sim.error === 'string')
+    // error type: just has error string
+    ? {
+      ...base,
+      error: sim.error,
+    }
+    // success type: might have a result (if invoking) and might have a
+    // restoration hint (if invoking AND some state is expired)
+    : {
```

Comment:
seems like a lot to inline, just for consideration, a little refactor:

```
{
  if (typeof sim.error === 'string')
      // error type: just has error string
      return {
        ...base,
        error: sim.error,
      }
  }
  
  return assembleSuccess(base, sim);
}

function assembleSuccess(base, sim) {
  // lines 153-187
}
```

Reply:
Great point, should have refactored this once it was stable. Done in 301329f8b1ebcd7c75843cb0ccc2dcf2a92282dd.